### PR TITLE
Upgrade to release/3.0

### PR DESCRIPTION
**Pull request by DavidBennettPIO**
I've got this upgraded to ffmpeg 3.0.9 !

Like discussed in #5 one day short of a year ago, you might want to make a `release/2.8` branch with the current master.

As an added bonus swresample and some other headers have been added.

I have also added some examples that are only a single `main` with step by step instructions with what order to do things as the ffmpeg ones are a bit hard to follow.
These compile and are fully functional so we can also use them as a sanity check.

Let me know what you think :smile:

### CODE REVIEW

**Comment by ljubobratovicrelja** [comment]
Wow this is nice, thank you!

There are a lot of changes, and I don't have time to check them out in a reasonably short period, so I'll trust you completely on this one. :)

I have branched out current master to a `release/2.8`, so you can merge this at your own will. I have added you as a collaborator on the project, so you should be able to tag a release after the merge. Hope you're ok with that, if not, please let me know.

**Comment by DavidBennettPIO** [comment]
Thanks for the invite!

I've been using my branch for a while for simple things without a problem, Also most of the common usage is covered by the examples (avfilter and avdevice are missing) so if I'm missing anything it's probably just some enum's from structs.

I'm in the process of upgrading it to 3.1 today so I'll hold out merging this until then as it might highlight some missing parts.

**Comment by ljubobratovicrelja** [comment]
> I'm in the process of upgrading it to 3.1 today so I'll hold out merging this until then as it might highlight some missing parts.

Even more awesome! 👍

**Comment by DavidBennettPIO** [comment]
A few minor things I picked up during the 3.1 update have been back ported. I'm merging this now.
